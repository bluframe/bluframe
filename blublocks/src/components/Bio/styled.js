// @flow

import styled, {
  type PropsWithTheme,
  type StyledComponent
} from "styled-components"
import Paragraph from "components/Typography/Paragraph"
import type { Theme } from "@bluframe/blublocks"
import colors from "styles/colors"
import mediaQueries from "styles/mediaQueries"

type Props = {|
  +color?: string
|}

export const Section: StyledComponent<
  {||},
  Theme,
  HTMLElement
> = styled.section`
  background: ${colors.white};
  border: 1px solid ${colors.gallery};
  column-gap: 20px;
  display: grid;
  font-family: ${({ theme }: PropsWithTheme<{||}, Theme>) => theme.font.family};
  grid-template-columns: 80px 1fr;
  padding: 10px;

  ${mediaQueries.tablet} {
    padding: 25px;
  }
`

export const Image: StyledComponent<{||}, Theme, HTMLImageElement> = styled.img`
  width: 80px;
`

export const Right: StyledComponent<{||}, Theme, HTMLDivElement> = styled.div`
  display: grid;
  row-gap: 10px;
`

export const Name: StyledComponent<{||}, Theme, HTMLHeadingElement> = styled.h4`
  color: ${colors.black};
  font-size: 18px;
  font-weight: 700;
  margin-bottom: 0;
  margin-top: 0;
`

export const Description: StyledComponent<
  Props,
  Theme,
  typeof Paragraph
> = styled(Paragraph)`
  font-size: 16px;
  line-height: 24px;
  text-align: justify;
`

export const Link: StyledComponent<Props, Theme, HTMLAnchorElement> = styled.a`
  color: ${({ color, theme }: PropsWithTheme<Props, Theme>) =>
    color || theme.palette.primary.main};
  font-weight: bold;

  &:hover {
    color: ${({ color, theme }: PropsWithTheme<Props, Theme>) =>
      color || theme.palette.primary.main};
  }
`
