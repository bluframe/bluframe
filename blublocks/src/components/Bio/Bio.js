// @flow

import { Name, Section, Top } from "./styled"
import type { ComponentProps } from "."
import Content from "components/Content"
import React from "react"

const Bio = ({
  components,
  description,
  image,
  name
}: ComponentProps): React$Node => (
  <Section>
    <Top>
      {image}
      <Name>{name}</Name>
    </Top>

    <Content components={components} content={description} />
  </Section>
)

export default Bio
