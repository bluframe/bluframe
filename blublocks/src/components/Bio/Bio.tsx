/* eslint-disable no-ternary, no-nested-ternary */

import { Name, NameAndSocialLinks, Section, Top } from "./styled"
import { ComponentProps } from "."
import Content from "components/Content"
import Link from "components/Typography/Link"
import SocialLinks from "components/SocialLinks"

const Bio = ({
  LinkComponent = Link,
  components,
  description,
  image,
  isAvatarCircle,
  isShowName = true,
  name,
  socialLinks,
  url
}: ComponentProps) => (
  <Section aria-label={name} role="region">
    <Top $isAvatarCircle={isAvatarCircle} $isShowName={isShowName}>
      {url ? <LinkComponent href={url}>{image}</LinkComponent> : image}
      {isShowName ? (
        <NameAndSocialLinks>
          {isShowName ? (
            url ? (
              <LinkComponent href={url}>
                <Name>{name}</Name>
              </LinkComponent>
            ) : (
              <Name>{name}</Name>
            )
          ) : null}
        </NameAndSocialLinks>
      ) : null}
    </Top>

    {socialLinks ? <SocialLinks {...socialLinks} /> : null}

    <Content components={components} content={description} />
  </Section>
)

export default Bio
