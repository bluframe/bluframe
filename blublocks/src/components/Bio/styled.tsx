import styled from "styled-components"
import Paragraph from "components/Typography/Paragraph"
import { Theme } from "styles/theme"
import colors from "styles/colors"
import mediaQueries from "styles/mediaQueries"

export const Section = styled.section`
  background: ${colors.white};
  border: 1px solid ${colors.gallery};
  display: grid;
  font-family: ${({ theme }: { theme: Theme }) => theme.font.family};
  padding: 10px;
  row-gap: 20px;

  ${mediaQueries.tablet} {
    padding: 25px;
  }
`

export const Image = styled.img`
  width: 80px;
`

export const Top = styled.div`
  align-items: center;
  column-gap: 20px;
  display: grid;
  grid-template-columns: 80px 1fr;
`

export const Name = styled.h4`
  color: ${colors.black};
  font-size: 18px;
  font-weight: 700;
  margin-bottom: 0;
  margin-top: 0;
`

export const Description = styled(Paragraph)`
  font-size: 16px;
  line-height: 24px;
  text-align: justify;
`

export const Link = styled.a`
  color: ${({ color, theme }: { color?: string; theme: Theme }) =>
    color || theme.palette.primary.main};
  font-weight: bold;

  &:hover {
    color: ${({ color, theme }: { color?: string; theme: Theme }) =>
    color || theme.palette.primary.main};
  }
`
