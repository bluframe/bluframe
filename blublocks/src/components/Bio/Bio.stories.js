// @flow

import Bio, { type Props } from "components/Bio"
import styled, { type StyledComponent } from "styled-components"
import React from "react"
import type { Story } from "@storybook/react"
import type { Theme } from "@bluframe/blublocks"
import mediaQueries from "styles/mediaQueries"

const stories = {
  component: Bio,
  title: "Components/Bio"
}

const Wrapper: StyledComponent<{||}, Theme, HTMLDivElement> = styled.div`
  width: 100%;

  ${mediaQueries.tablet} {
    width: 600px;
  }
`

const Template = (args: Props) => (
  <Wrapper>
    <Bio {...args} />
  </Wrapper>
)

export const Default: Story<Props> = Template.bind({})
Default.args = {
  avatar:
    "https://0.gravatar.com/avatar/c88fdd33c5a3a417782e94a0774cee15?s=96&#038;d=mm&#038;r=g",
  description:
    "Edoardo L'Astorina has 15 years of experience in software development. Edoardo has had a major role in the new [Transport for London](https://tfl.gov.uk) site and led the development of the new [Royal Opera House](https://roh.org.uk) site. Edoardo has developed sites and apps for [Intuit](https://intuit.com), [Stint](https://stint.co), [JPC](https://thinkjpc.com), [The Crocodile](//thecroc.com) and [Miura](https://miura.gi) Edoardo started [Blu Frame](https://blufra.me) to help companies develop sites that stand out, load fast and are easy for users to access Edoardo is passionate about risotto, Terrence Malick movies, Oasis songs and rowing",
  name: "Edoardo L'Astorina"
}

export const Color: Story<Props> = Template.bind({})
Color.args = {
  avatar:
    "https://0.gravatar.com/avatar/c88fdd33c5a3a417782e94a0774cee15?s=96&#038;d=mm&#038;r=g",
  color: "#FF7D14",
  description:
    "Edoardo L'Astorina has 15 years of experience in software development. Edoardo has had a major role in the new [Transport for London](https://tfl.gov.uk) site and led the development of the new [Royal Opera House](https://roh.org.uk) site. Edoardo has developed sites and apps for [Intuit](https://intuit.com), [Stint](https://stint.co), [JPC](https://thinkjpc.com), [The Crocodile](//thecroc.com) and [Miura](https://miura.gi) Edoardo started [Blu Frame](https://blufra.me) to help companies develop sites that stand out, load fast and are easy for users to access Edoardo is passionate about risotto, Terrence Malick movies, Oasis songs and rowing",
  name: "Edoardo L'Astorina"
}

export default stories
