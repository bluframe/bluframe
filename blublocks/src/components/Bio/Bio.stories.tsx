import Bio from "components/Bio"
import { StoryObj } from "@storybook/react"
import mediaQueries from "styles/mediaQueries"
import styled from "styled-components"

const meta = {
  component: Bio,
  title: "Components/Bio"
}

const Wrapper = styled.div`
  width: 100%;

  ${mediaQueries.tablet} {
    width: 600px;
  }
`

type Story = StoryObj<typeof Bio>

export const Default: Story = {
  args: {
    avatar:
      "https://0.gravatar.com/avatar/c88fdd33c5a3a417782e94a0774cee15?s=96&#038;d=mm&#038;r=g",
    description:
      "Edoardo L'Astorina has 15 years of experience in software development. Edoardo has had a major role in the new [Transport for London](https://tfl.gov.uk) site and led the development of the new [Royal Opera House](https://roh.org.uk) site. Edoardo has developed sites and apps for [Intuit](https://intuit.com), [Stint](https://stint.co), [JPC](https://thinkjpc.com), [The Crocodile](//thecroc.com) and [Miura](https://miura.gi) Edoardo started [Blu Frame](https://blufra.me) to help companies develop sites that stand out, load fast and are easy for users to access Edoardo is passionate about risotto, Terrence Malick movies, Oasis songs and rowing",
    name: "Edoardo L'Astorina"
  },
  render: (args: any) => (
    <Wrapper>
      <Bio {...args} />
    </Wrapper>
  )
}

export const Color: Story = {
  args: {
    avatar:
      "https://0.gravatar.com/avatar/c88fdd33c5a3a417782e94a0774cee15?s=96&#038;d=mm&#038;r=g",
    color: "#FF7D14",
    description:
      "Edoardo L'Astorina has 15 years of experience in software development. Edoardo has had a major role in the new [Transport for London](https://tfl.gov.uk) site and led the development of the new [Royal Opera House](https://roh.org.uk) site. Edoardo has developed sites and apps for [Intuit](https://intuit.com), [Stint](https://stint.co), [JPC](https://thinkjpc.com), [The Crocodile](//thecroc.com) and [Miura](https://miura.gi) Edoardo started [Blu Frame](https://blufra.me) to help companies develop sites that stand out, load fast and are easy for users to access Edoardo is passionate about risotto, Terrence Malick movies, Oasis songs and rowing",
    name: "Edoardo L'Astorina"
  }
}

export const ImageComponent: Story = {
  args: {
    avatar: (
      <img
        alt="Edoardo L'Astorina"
        src="https://0.gravatar.com/avatar/c88fdd33c5a3a417782e94a0774cee15?s=96&#038;d=mm&#038;r=g"
      />
    ),
    description:
      "Edoardo L'Astorina has 15 years of experience in software development. Edoardo has had a major role in the new [Transport for London](https://tfl.gov.uk) site and led the development of the new [Royal Opera House](https://roh.org.uk) site. Edoardo has developed sites and apps for [Intuit](https://intuit.com), [Stint](https://stint.co), [JPC](https://thinkjpc.com), [The Crocodile](//thecroc.com) and [Miura](https://miura.gi) Edoardo started [Blu Frame](https://blufra.me) to help companies develop sites that stand out, load fast and are easy for users to access Edoardo is passionate about risotto, Terrence Malick movies, Oasis songs and rowing",
    name: "Edoardo L'Astorina"
  }
}

export default meta
