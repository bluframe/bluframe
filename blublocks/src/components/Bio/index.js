// @flow

import type { BioProps, ContentComponents } from "@bluframe/blublocks"
import { Description, Image, Link } from "./styled"
import Bio from "./Bio"
import React from "react"
import { prepareComponent } from "@bluframe/grapple"

export type Props = {|
  ...BioProps
|}

export type ComponentProps = {|
  ...$Diff<BioProps, {| +avatar: React$Node | string, +color?: string |}>,
  +components: ContentComponents,
  +image: React$Node
|}

export const usePrepareComponent = ({
  avatar,
  color,
  ...props
}: Props): ComponentProps => {
  // If we ever need our own shared overrides
  const components: ContentComponents = {
    // eslint-disable-next-line react/no-multi-comp, react/display-name, react/no-unstable-nested-components, id-length
    a: ({ children, href, ...componentProps }) => (
      <Link color={color} href={href} {...componentProps}>
        {children}
      </Link>
    ),
    // eslint-disable-next-line react/no-multi-comp, react/display-name, react/no-unstable-nested-components, id-length
    p: (componentProps) => <Description {...componentProps} />
  }

  let image = avatar

  if (typeof avatar === "string") {
    image = <Image alt={props.name} src={avatar} />
  }

  return {
    ...props,
    components,
    image
  }
}

export default (prepareComponent<Props, ComponentProps>(usePrepareComponent)(
  Bio
): React$ComponentType<Props>)
