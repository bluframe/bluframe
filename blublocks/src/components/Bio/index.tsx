"use client"

import { Description, Image, Link } from "./styled"
import Bio from "./Bio"
import { ContentComponents } from "components/Content"
import React from "react"
import { Props as SocialLinksProps } from "components/SocialLinks"
import { prepareComponent } from "@bluframe/grapple"

export interface Props {
  LinkComponent?: React.JSXElementConstructor<any>
  avatar: React.ReactNode | string
  color?: string
  components?: ContentComponents
  description: string
  isAvatarCircle?: boolean
  isShowName?: boolean
  name: string
  socialLinks?: SocialLinksProps
  url?: string
}

export interface ComponentProps extends Omit<Props, "avatar" | "color"> {
  components: ContentComponents
  image: React.ReactNode
}

export const usePrepareComponent = ({
  avatar,
  color,
  components: overrideComponents,
  ...props
}: Props): ComponentProps => {
  // If we ever need our own shared overrides
  const components: ContentComponents = {
    // eslint-disable-next-line id-length
    a: ({ children, href, node, ...componentProps }) => (
      <Link color={color} href={href} {...componentProps}>
        {children}
      </Link>
    ),
    // eslint-disable-next-line id-length
    p: ({ node, ...componentProps }) => <Description {...componentProps} />
  }

  let image = avatar

  if (typeof avatar === "string") {
    image = <Image alt={props.name} src={avatar} />
  }

  return {
    ...props,
    components: {
      ...components,
      ...overrideComponents
    },
    image
  }
}

export default prepareComponent<Props, ComponentProps>(usePrepareComponent)(Bio)
