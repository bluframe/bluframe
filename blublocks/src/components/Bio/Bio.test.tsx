/* eslint-disable max-lines-per-function */

import { render, screen } from "tests"
import Bio from "./Bio"
import { ComponentProps } from "."

describe("Bio", () => {
  let props: ComponentProps

  beforeEach(() => {
    props = {
      components: {},
      description:
        "Edoardo L'Astorina has 15 years of experience in software development. Edoardo has had a major role in the new [Transport for London](https://tfl.gov.uk) site and led the development of the new [Royal Opera House](https://roh.org.uk) site. Edoardo has developed sites and apps for [Intuit](https://intuit.com), [Stint](https://stint.co), [JPC](https://thinkjpc.com), [The Crocodile](//thecroc.com) and [Miura](https://miura.gi) Edoardo started [Blu Frame](https://blufra.me) to help companies develop sites that stand out, load fast and are easy for users to access Edoardo is passionate about risotto, Terrence Malick movies, Oasis songs and rowing",
      image: (
        <img
          alt="Edoardo L'Astorina"
          src="https://0.gravatar.com/avatar/c88fdd33c5a3a417782e94a0774cee15?s=96&#038;d=mm&#038;r=g"
        />
      ),
      name: "Edoardo L'Astorina"
    }
  })

  it("renders", () => {
    render(<Bio {...props} />)

    expect(
      screen.getByRole("region", { name: "Edoardo L'Astorina" })
    ).toBeInTheDocument()

    expect(
      screen.getByRole("img", { name: "Edoardo L'Astorina" })
    ).toBeInTheDocument()
    expect(
      screen.getByRole("heading", { level: 4, name: "Edoardo L'Astorina" })
    ).toBeInTheDocument()
  })

  it("renders social links", () => {
    props.socialLinks = {
      instagram: "mightychroma"
    }

    render(<Bio {...props} />)

    expect(
      screen.getByRole("list", { name: "Social Links" })
    ).toBeInTheDocument()
    expect(
      screen.getByRole("listitem", { name: "Instagram" })
    ).toBeInTheDocument()
    expect(screen.getByTitle("Instagram")).toBeInTheDocument()
    expect(screen.getByRole("link", { name: "Instagram" })).toBeInTheDocument()
    expect(screen.getByRole("link", { name: "Instagram" })).toHaveAttribute(
      "href",
      "https://instagram.com/mightychroma"
    )
    expect(screen.getByRole("link", { name: "Instagram" })).toHaveAttribute(
      "target",
      "__blank"
    )
  })
})
