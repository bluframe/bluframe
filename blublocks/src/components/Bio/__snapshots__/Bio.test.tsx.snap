// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio renders 1`] = `
<section
  class="sc-gtcySc dkCDlJ"
>
  <div
    class="sc-hKVqoC lfZKxd"
  >
    <img
      alt="Edoardo L'Astorina"
      src="https://0.gravatar.com/avatar/c88fdd33c5a3a417782e94a0774cee15?s=96&d=mm&r=g"
    />
    <h4
      class="sc-eCQicT ehmVUK"
    >
      Edoardo L'Astorina
    </h4>
  </div>
  <span
    class="sc-iCEzBS ddoKtl"
  >
    <div>
      Edoardo L'Astorina has 15 years of experience in software development. Edoardo has had a major role in the new [Transport for London](https://tfl.gov.uk) site and led the development of the new [Royal Opera House](https://roh.org.uk) site. Edoardo has developed sites and apps for [Intuit](https://intuit.com), [Stint](https://stint.co), [JPC](https://thinkjpc.com), [The Crocodile](//thecroc.com) and [Miura](https://miura.gi) Edoardo started [Blu Frame](https://blufra.me) to help companies develop sites that stand out, load fast and are easy for users to access Edoardo is passionate about risotto, Terrence Malick movies, Oasis songs and rowing
    </div>
  </span>
</section>
`;
